src/bindings/python/src/pyopenvino/graph/passes/pattern_ops.cpp

```diff
@@ -537,56 +536,140 @@ static void reg_pattern_optional(py::module m) {
     )");
 
     optional_type.def(
-        py::init([](const std::vector<std::string>& type_names, const ov::Output<ov::Node>& input) {
-            return std::make_shared<ov::pass::pattern::op::Optional>(get_types(type_names), input, nullptr);
+        py::init([](const std::vector<std::string>& type_names, const ov::OutputVector& inputs) {
+            return std::make_shared<ov::pass::pattern::op::Optional>(get_types(type_names), inputs, nullptr);
         }),
         py::arg("type_names"),
-        py::arg("input"),
+        py::arg("inputs"),
         R"(
         Create Optional with the given node type and input node.
 
         :param type_names: node type. For example: ["opset8.Abs", "opset8.Relu"]
         :type type_names: List[str]
 
-        :param input: input node's output.
-        :type input: openvino.runtime.Output
+        :param inputs: input node's output vector.
+        :type inputs: List[openvino.runtime.Output]
     )");
 
-    optional_type.def(
-        py::init([](const std::vector<std::string>& type_names, const std::shared_ptr<ov::Node>& input) {
-            return std::make_shared<ov::pass::pattern::op::Optional>(get_types(type_names), input, nullptr);
-        }),
-        py::arg("type_names"),
-        py::arg("input"),
-        R"(
+    optional_type.def(py::init([](const std::vector<std::string>& type_names, const ov::NodeVector& inputs) {
+                          return std::make_shared<ov::pass::pattern::op::Optional>(get_types(type_names),
+                                                                                   ov::as_output_vector(inputs),
+                                                                                   nullptr);
+                      }),
+                      py::arg("type_names"),
+                      py::arg("inputs"),
+                      R"(
         Create Optional with the given node type and input node.
 
         :param type_names: node type. For example: ["opset8.Abs", "opset8.Relu"]
         :type type_names: List[str]
 
-        :param input: input node
-        :type input: openvino.runtime.Node
+        :param inputs: input node vector
+        :type inputs: List[openvino.runtime.Node]
+    )");
+
+    optional_type.def(py::init([](const std::vector<std::string>& type_names, const Predicate& predicate) {
+                          return std::make_shared<ov::pass::pattern::op::Optional>(get_types(type_names),
+                                                                                   ov::OutputVector{},
+                                                                                   predicate);
+                      }),
+                      py::arg("type_names"),
+                      py::arg("predicate"),
+                      R"(
+        Create Optional with the given node type and predicate.
```

Comment:
```suggestion
        Create Optional with the given node type and predicate.

```
same about the empty line